sbom-utility-scripts/scripts/index-image-sbom-script/index_image_sbom_script.py

```diff
@@ -0,0 +1,196 @@
+#!/usr/bin/env python3
+import argparse
+import json
+from datetime import datetime, timezone
+from pathlib import Path
+from typing import Optional, Any
+from dataclasses import dataclass
+from uuid import uuid4
+
+from packageurl import PackageURL
+
+
+@dataclass
+class Image:
+    repository: str
+    name: str
+    digest: str
+    tag: str
+    arch: Optional[str]
+
+    @staticmethod
+    def from_image_index_url_and_digest(
+        image_url_and_tag: str,
+        image_digest: str,
+        arch: Optional[str] = None,
+    ) -> "Image":
+
+        repository, tag = image_url_and_tag.rsplit(":", 1)
+        _, name = repository.rsplit("/", 1)
+        return Image(
+            repository=repository,
+            name=name,
+            digest=image_digest,
+            tag=tag,
+            arch=arch,
+        )
+
+    @property
+    def digest_algo(self) -> str:
+        algo, _ = self.digest.split(":")
+        return algo.upper()
+
+    @property
+    def digest_hex_val(self) -> str:
+        _, val = self.digest.split(":")
+        return val
+
+    def purls(self, index_digest: Optional[str] = None) -> list[str]:
+        ans = []
+        if index_digest and self.arch:
+            ans.append(
+                PackageURL(
+                    type="oci",
+                    name=self.name,
+                    version=index_digest,
+                    qualifiers={"arch": self.arch, "repository_url": self.repository},
+                ).to_string()
+            )
+        ans.append(
+            PackageURL(
+                type="oci", name=self.name, version=self.digest, qualifiers={"repository_url": self.repository}
+            ).to_string()
+        )
+        return ans
+
+    def propose_spdx_id(self) -> str:
+        return f"SPDXRef-{self.digest_hex_val}"
+
+
+def create_package(image: Image, spdxid: Optional[str] = None, image_index_digest: Optional[str] = None) -> dict:
+    return {
+        "SPDXID": image.propose_spdx_id() if not spdxid else spdxid,
```

Comment:
The conversation [here](https://github.com/konflux-ci/build-tasks-dockerfiles/pull/163#discussion_r1766359666) has led me to believe that the format of SPDXID is a decision with long-term impact. In every PR that adds SPDX support, there's a different approach taken for the SPDXID
* this one
* https://github.com/konflux-ci/build-tasks-dockerfiles/pull/163
* https://github.com/containerbuildsystem/cachi2/pull/608

It would be great if you could sync with @midnightercz and the stakeholders (Red Hat prodsec?) on what the SPDXID should be

Reply:
I have tried to search for guidelines from ProdSec and I have found [this](https://github.com/RedHatProductSecurity/security-data-guidelines/blob/main/docs/sbom.md#packages-and-relationships). It states that the `SPDXID` must be unique *within* the SBOM, but I agree that we should double-check with them so we don't prepare any traps for ourselves in the future.